db/fixtures/users.yml

```
@@ -864,3 +864,531 @@ thuynga: # アイコンが小さい画像のユーザー
+  name_kana: マルマル シャイン18
+  twitter_account:
+  facebook_url:
+  blog_url:
+  company: company1
+  description: "マルマル企業の社員18"
+  course: course1
+  job: office_worker
+  os: mac
+  experience: rails
+  organization: 株式会社フィヨルド
+  admin: true
+  mentor: true
+  github_collaborator: true
+  unsubscribe_email_token: MnC-Dv3HrNLs7WmfE975qA
+  updated_at: "2014-01-01 00:00:02"
+  created_at: "2014-01-01 00:00:02"
+
+marumarushain19:
+  login_name: marumarushain19
+  email: marumarushain19@gmail.com
+  crypted_password: $2a$10$n/xv4/1luueN6plzm2OyDezWlZFyGHjQEf4hwAW1r3k.lCm0frPK. # testtest
+  salt: zW3kQ9ubsxQQtzzzs4ap
+  name: marumarushain19
+  name_kana: マルマル シャイン19
+  twitter_account:
+  facebook_url:
+  blog_url:
+  company: company1
+  description: "マルマル企業の社員19"
+  course: course1
+  job: office_worker
+  os: mac
+  experience: rails
+  organization: 株式会社フィヨルド
+  admin: true
+  mentor: true
+  github_collaborator: true
+  unsubscribe_email_token: MnC-Dv3HrNLs7WmfE975qA
+  updated_at: "2014-01-01 00:00:02"
+  created_at: "2014-01-01 00:00:02"
+
+marumarushain20:
+  login_name: marumarushain20
+  email: marumarushain20@gmail.com
+  crypted_password: $2a$10$n/xv4/1luueN6plzm2OyDezWlZFyGHjQEf4hwAW1r3k.lCm0frPK. # testtest
+  salt: zW3kQ9ubsxQQtzzzs4ap
+  name: marumarushain20
+  name_kana: マルマル シャイン20
+  twitter_account:
+  facebook_url:
+  blog_url:
+  company: company1
+  description: "マルマル企業の社員20"
+  course: course1
+  job: office_worker
+  os: mac
+  experience: rails
+  organization: 株式会社フィヨルド
+  admin: true
+  mentor: true
+  github_collaborator: true
+  unsubscribe_email_token: MnC-Dv3HrNLs7WmfE975qA
+  updated_at: "2014-01-01 00:00:02"
+  created_at: "2014-01-01 00:00:02"
+
+marumarushain21:
+  login_name: marumarushain21
+  email: marumarushain21@gmail.com
+  crypted_password: $2a$10$n/xv4/1luueN6plzm2OyDezWlZFyGHjQEf4hwAW1r3k.lCm0frPK. # testtest
+  salt: zW3kQ9ubsxQQtzzzs4ap
+  name: marumarushain20
+  name_kana: マルマル シャイン21
+  twitter_account:
+  facebook_url:
+  blog_url:
+  company: company1
+  description: "マルマル企業の社員21"
+  course: course1
+  job: office_worker
+  os: mac
+  experience: rails
+  organization: 株式会社フィヨルド
+  admin: true
+  mentor: true
+  github_collaborator: true
+  unsubscribe_email_token: MnC-Dv3HrNLs7WmfE975qA
+  updated_at: "2014-01-01 00:00:02"
+  created_at: "2014-01-01 00:00:02"
+
+marumarushain22:
+  login_name: marumarushain22
+  email: marumarushain22@gmail.com
+  crypted_password: $2a$10$n/xv4/1luueN6plzm2OyDezWlZFyGHjQEf4hwAW1r3k.lCm0frPK. # testtest
+  salt: zW3kQ9ubsxQQtzzzs4ap
+  name: marumarushain22
+  name_kana: マルマル シャイン22
+  twitter_account:
+  facebook_url:
+  blog_url:
+  company: company1
+  description: "マルマル企業の社員22"
+  course: course1
+  job: office_worker
+  os: mac
+  experience: rails
+  organization: 株式会社フィヨルド
+  admin: true
+  mentor: true
+  github_collaborator: true
+  unsubscribe_email_token: MnC-Dv3HrNLs7WmfE975qA
+  updated_at: "2014-01-01 00:00:02"
+  created_at: "2014-01-01 00:00:02"
```

```suggestion
<% 1.upto(22) do |i| %>
marumarushain<%= i %>:
  login_name: marumarushain<%= i %>
  email: marumarushain<%= i %>@fjord.jp
  crypted_password: $2a$10$n/xv4/1luueN6plzm2OyDezWlZFyGHjQEf4hwAW1r3k.lCm0frPK. # testtest
  salt: zW3kQ9ubsxQQtzzzs4ap
  name: marumarushain<%= i %>
  name_kana: マルマル シャイン<%= i %>
  twitter_account:
  facebook_url:
  blog_url:
  company: company1
  description: マルマル企業の社員<%= i %>
  course: course1
  job: office_worker
  os: mac
  experience: rails
  organization: 株式会社フィヨルド
  admin: true
  mentor: true
  github_collaborator: true
  unsubscribe_email_token: MnC-Dv3HrNLs7WmfE975qA
  updated_at: "2014-01-01 00:00:02"
  created_at: "2014-01-01 00:00:02"
<% end %>
```
- ループでまとめてあげた方が、可読性やメンテナンスの容易さが向上するかと思います。
- `gmail.com`のような実在するドメインは、万が一の事故を防ぐため、使用を避けた方が無難かと思いました。
- こちらのデータを追加なさった目的が分かりませんでした（ページネーションの確認でしょうか）。目的によっては、テストを作成した方が目的に適っているかもしれないと思いました（駒形さんにご確認頂くのが良いかと思います）。

---

- これはページネーションの確認のために追加しました。
- たしかにループでまとめるべきですね！
- たしかに必要でなければ実在するドメインはさけるべきですね...。
- テストを作成するべきか駒形さんに確認したいと思います。

---

@komagata 
お疲れ様です。
user一覧のページネーションの確認をしたいのですが、この場合、どうするべきでしょうか？

1. テストを作成し、テストで確認すべき。
2. テストは不要。seedにデータを入れて直接確認できればok(現在の実装)
3. テスト、seedともに必要
4. その他

seedにデータがあれば今後デモするときに便利なのと、他のページネーションのあるQ&Aやユーザー一覧でページネーションに関するtestをしていないことから、2.で実装しました。
見解をいただけると幸いです。よろしくお願いいたします。

---

@naomichi-h 

2

です〜

ただご指摘受けているようにseedはeachで書くのがいいかもです。

ページネーションのテストをする必要がなければテスト環境のfixtureはいらないです。（不要なfixtureがあるとテストが遅くなるので）

development環境のseedが欲しい理由は、デザイン確認と、よくある形にしておくことで他の開発者も含めて使い勝手の確認や不具合の発見が容易になるからです〜